lms/djangoapps/course_api/blocks/api.py

```diff
@@ -52,6 +53,7 @@ def get_blocks(
     # create ordered list of transformers, adding BlocksAPITransformer at end.
     transformers = BlockStructureTransformers()
     include_special_exams = False
+    include_completion = requested_fields is not None and 'completion' in requested_fields
```

Comment:
*Future note to self (and others):* If there are any great ideas of making this automatically pluggable as well, welcoming ideas/contributions.